resfulservice/src/controllers/curationController.js

```diff
@@ -16,40 +18,42 @@ exports.curateXlsxSpreadsheet = async (req, res, next) => {
   if (!req.files?.uploadfile) {
     return next(errorWriter(req, 'Material template files not uploaded', 'curateXlsxSpreadsheet', 400));
   }
-
-  const regex = /master_template.xlsx$/gi;
+  const regex = /(?=.*?(master_template))(?=.*?(.xlsx)$)/gi;
   const xlsxFile = req.files.uploadfile.find((file) => regex.test(file?.path));
 
   if (!xlsxFile) {
     return next(errorWriter(req, 'Master template xlsx file not uploaded', 'curateXlsxSpreadsheet', 400));
   }
 
-  if (!query.dataset) {
-    return next(errorWriter(req, 'Missing dataset ID in query', 'curateXlsxSpreadsheet', 400));
-  }
-
   try {
-    const [validList, storedCurations, datasets] = await Promise.all([
+    const [validList, storedCurations] = await Promise.all([
       XlsxCurationList.find({}, null, { lean: true }),
-      CuratedSamples.find({ user: user._id }, { object: 1 }, { lean: true }),
-      DatasetId.findOne({ _id: query.dataset })
+      CuratedSamples.find({ user: user._id }, { object: 1 }, { lean: true })
     ]);
 
-    if (!datasets) {
-      return next(errorWriter(req, `A sample must belong to a dataset. Dataset ID: ${query.dataset ?? null} not found`, 'curateXlsxSpreadsheet', 404));
-    }
-
     const validListMap = generateCurationListMap(validList);
-    const result = await this.createMaterialObject(xlsxFile.path, BaseSchemaObject, validListMap, req.files.uploadfile);
-
+    const processedFiles = [];
+    const result = await this.createMaterialObject(xlsxFile.path, BaseSchemaObject, validListMap, req.files.uploadfile, processedFiles);
+    if (result?.count && req?.isParentFunction) return { errors: result.errors };
     if (result?.count) return res.status(400).json({ errors: result.errors });
 
     const curatedAlready = storedCurations.find(
-      object => object?.['data origin']?.Title === result?.['data origin']?.Title &&
-      object?.['data origin']?.PublicationType === result?.['data origin']?.PublicationType);
+      object => object?.DATA_SOURCE?.Citation?.CommonFields?.Title === result?.DATA_SOURCE?.Citation?.CommonFields?.Title &&
+      object?.DATA_SOURCE?.Citation?.CommonFields?.PublicationType === result?.DATA_SOURCE?.Citation?.CommonFields?.PublicationType);
 
-    if (curatedAlready) return next(errorWriter(req, 'This has already been curated', 'curateXlsxSpreadsheet', 409));
+    if (curatedAlready) return next(errorWriter(req, 'This had been curated already', 'curateXlsxSpreadsheet', 409));
```

Comment:
I'm not sure if I agree with this change

Reply:
I agree with that part, I just meant the wording/phrasing change

Reply:
@aswallace DM me your suggestion for rephrasing and I'll update it in the next PR